feat: Add LIBRA benchmark for long-context evaluation

karimovaSvetlana opened the pull request:
### **Description**  
This PR adds support for the **LIBRA** benchmark (`Long Input Benchmark for Russian Analysis`) to `lm-harness`, enabling evaluation of LLMs on long-context tasks. The benchmark includes 21 datasets across 4 complexity groups, adapted from popular Russian-language tasks.  

#### **Key Changes**:  
1. **New Task Groups**:  
   - `libra_simple_information_retrieval`  
   - `libra_question_answering_and_multiple_choice`  
   - `libra_multi_hop_question_answering`  
   - `libra_complex_reasoning_and_mathematical_problems`  

2. **Datasets**:  
   - Implements all 21 LIBRA tasks (e.g., `passkey`, `ru_babilong_qa1-5`, `ru_2wikimultihopqa`, `ru_gsm100`).  
   - Supports `--apply_chat_template` for consistent prompting.  

3. **Documentation**:  
   - Added README section with:  
     - Dataset overview, paper citation, and homepage.  
     - Usage examples (see below).  
     - Task variants and evaluation groups.  

#### **Usage**:  
```bash
lm_eval --model hf \
        --model_args pretrained=EleutherAI/gpt-j-6B \
        --tasks libra_simple_information_retrieval \
        --apply_chat_template \
        --log_samples \
        --output_path libra_results
```

#### **Checklist**:  
- [x] Referenced original LIBRA paper ([arXiv:2408.02439](https://arxiv.org/abs/2408.02439)).  
- [x] Confirmed task alignment with the reference implementation.  
- [x] Documented all variants in README (see `Groups/Tasks` section).  
- [x] Denoted "Main" variants.

#### **Testing**:  
- Verified against `gpt-j-6B` on CPU/GPU.  
- Cross-checked outputs with HuggingFace dataset samples.  

### **Motivation**  
LIBRA fills a critical gap in non-English long-context evaluation, providing standardized metrics for Russian LLMs. Integration into `lm-harness` ensures reproducibility and compatibility with existing workflows.  

### **References**  
- Paper: [Churin et al., 2024](https://arxiv.org/abs/2408.02439)  
- Dataset: [HuggingFace](https://huggingface.co/datasets/ai-forever/LIBRA)  

---

### **Notes for Reviewers**:  
1. **Chat Templates**: All tasks require `--apply_chat_template` (LIBRA’s default prompt format).  
2. **Task Variants**: Some tasks (e.g., `passkey_with_librusec`) extend base datasets; differences are documented.  
3. **Multilingual Impact**: Benchmarks are Russian-only but designed for cross-lingual comparisons.  

Let me know if any adjustments are needed!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2943) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2943) before we can accept your contribution.<br/><hr/>**Светлана Каримова** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2943) it.</sub>

baberabb commented:
Hi! Thanks for the PR! LGTM, except can you add an entry to `lm_eval/tasks/ReadMe.md`. And fix the formatting:
```
pip install pre-commit
pre-commit install
pre-commit run --all-files
```

karimovaSvetlana commented:
Great, thanks for the feedback!

Added an entry to `lm_eval/tasks/ReadMe.md` and fixed the formatting by running `pre-commit run --all-files`. New commit pushed to the PR.

baberabb commented:
Hi! Test is failing due to 
```
ERROR tests/test_tasks.py - datasets.exceptions.DatasetNotFoundError: Dataset 'ai-forever/libra-v2' doesn't exist on the Hub or cannot be accessed.
```
Is this the correct dataset?

karimovaSvetlana commented:
Hello!
Could you please define if everything okay with PR and can it be merged?

Thank you!

baberabb commented:
thanks for the PR!
